vNext/extensions/applicationinsights-analytics-js/src/JavaScriptSDK/ApplicationInsights.ts

```diff
@@ -609,8 +612,15 @@ export class ApplicationInsights implements IAppInsights, ITelemetryPlugin, IApp
                     _self._properties.context.telemetryTrace.name = window.location.pathname;
                 }
                 setTimeout(() => {
+                    if (this._urlArr.length == 2) {
+                        var prevUrl = this._urlArr[1];
+                        this._urlArr[0] = prevUrl;
+                        this._urlArr[1] = window.location.href;
```

Comment:
Because this is wrapped in a `setTimeout`, this wont work if the URL changes earlier than `autoRoutePVDelay`. Is it possible to send in the previous url via the event emitters?

Reply:
yeah that makes sense, update in locationchange event before settimeout